gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/GradingServiceImpl.java

```diff
@@ -655,21 +658,28 @@ public Map<String,String> transferGradebook(final GradebookInformation gradebook
 
         if (!categories.isEmpty() && !Objects.equals(gradebookInformation.getCategoryType(), GradingConstants.CATEGORY_TYPE_NO_CATEGORY)) {
 
-            // migrate the categories with assignments
+            Integer toCategoryType = gradebook.getCategoryType();
+
+            // conditionally migrate the categories with assignments
             categories.forEach(c -> {
 
                 assignments.forEach(a -> {
-                    String taskName = isGradebookGroupEnabled(fromContext)
-                            ? a.getName().substring(a.getName().indexOf('-') + 1)
+                    int dash = a.getName().lastIndexOf('-');
+                    String taskName = isGradebookGroupEnabled(fromContext) && dash >= 0
+                            ? a.getName().substring(dash + 1)
                             : a.getName();
 
-                    if (StringUtils.equals(c.getName(), a.getCategoryName())) {
+                    if (copySettings && StringUtils.equals(c.getName(), a.getCategoryName())
+			&& !Objects.equals(toCategoryType, GradingConstants.CATEGORY_TYPE_NO_CATEGORY)) {
 
                         if (!categoriesCreated.containsKey(c.getName())) {
                             // create category
                             Long categoryId = null;
                             try {
-                                categoryId = createCategory(gradebook.getId(), c.getName(), c.getWeight(), c.getDropLowest(),
+				    Double weight = Objects.equals(toCategoryType, GradingConstants.CATEGORY_TYPE_WEIGHTED_CATEGORY)
+						 ? c.getWeight()
+						 : Double.valueOf(0.0);
+                                categoryId = createCategory(gradebook.getId(), c.getName(), weight, c.getDropLowest(),
```

Comment:
why did you need to make this change? are you fixing a bug where formerly-weighted categories still had a weight on copy?!